### PR TITLE
Add API for gitignore templates

### PR DESCRIPTION
**Pull request by JakobDev**
This implements the [Gitignores template API of GitHub](https://docs.github.com/en/rest/gitignore?apiVersion=2022-11-28) in Gitea

### CODE REVIEW

File: routers/api/v1/misc/gitignore.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	//   in: path
+	//   description: name of the template
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/GitignoreTemplateInfo"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	name := ctx.Params("name")
```
Comment (lunny) [inline routers/api/v1/misc/gitignore.go]: The path needs to be clean

**Review by lunny — CHANGES_REQUESTED** [review]

File: routers/api/v1/misc/gitignore.go (hunk 1 above)
Comment (JakobDev) [reply routers/api/v1/misc/gitignore.go]: What did you mean?

Comment (lunny) [reply routers/api/v1/misc/gitignore.go]: name maybe `../../path`, it's a security problem.

Comment (JakobDev) [reply routers/api/v1/misc/gitignore.go]: What is the preferred function to do this? I couldn't find a example in the other API code.

Comment (wxiaoguang) [reply routers/api/v1/misc/gitignore.go]: use `util.PathJoinRelX()` to process the user inputed name, then LGTM.

----

Update: after some refactoring PRs, the `options.Gitignore` is safe enough now, so even without "path cleaning", the call to options.Gitignore won't cause security problems.

However, I would prefer to clean this name, to make future reviewers easy to understand that this is safe enough, and the "name" will be returned by `GitignoreTemplateInfo`, so a cleaned name seems better.

**Review by wxiaoguang — APPROVED** [review]

File: routers/api/v1/misc/gitignore.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	// summary: Returns a list of all gitignore templates
+	// produces:
+	// - application/json
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/GitignoreTemplateList"
+	gitignoreList := make([]string, len(repo_module.Gitignores))
+
+	for position, currentGitignore := range repo_module.Gitignores {
+		gitignoreList[position] = util.PathJoinRelX(currentGitignore)
```
Comment (wxiaoguang) [inline routers/api/v1/misc/gitignore.go]:
```suggestion
		gitignoreList[position] = currentGitignore
```

**Review by wxiaoguang — APPROVED** [review]
As the review in another PR.

<details>

`util.PathJoinRelX` means "clean the path to a relative path", it's only needed for uncontrolable user input. It has no affect on Gitea internal data like `Gitignores`.

</details>

**Review by lunny — APPROVED** [review]

**Pull request merged**